src/getAuthorsWithSlackIds.js

```diff
@@ -0,0 +1,52 @@
+const core = require('@actions/core');
+const { WebClient } = require('@slack/web-api');
+
+/**
+ * Enhance authors with Slack IDs matching their email addresses.
+ *
+ * If the whole function fails, or some emails cannot be matched, the original authors are returned.
+ *
+ * @param {string} slackToken
+ * @param {array<{ name: string, email: string }>} authors
+ * @returns {Promise<array<{ name: string, email: string, slackId?: string }>>}
+ */
+async function getAuthorsWithSlackIds(slackToken, authors) {
+    if (!authors.length) {
+        core.info('No authors to fetch Slack IDs for');
```

Comment:
Nit, maybe a warning here.

Reply:
This is actually fine... `authors` can be empty... either no commits in the changelog, or the changelog is generated from the PR title or description.

Arguably, we should not be calling this function then at all... I can update it one level up, but the current approach feels fine.

Also, the condition is actually not needed, but in case there are no `authors`, it skips fetching Slack users.